txpipe/covariance.py

```diff
@@ -541,3 +543,216 @@ def save_outputs(self, two_point_data, cov):
+            cl_file.save_fits(output_filename, overwrite=True)
+            print("Saved power spectra with its Gaussian covariance")
+
+    def get_workspaces_dict(self, cl_file, masks_names):
+        cache_dir = cl_file.metadata['cache_dir']
+        cache = self.load_workspace_cache(cache_dir)
+        if cache == {}:
+            return {}
+
+        hashes = {}
+        masks_names_list = list(masks_names.values())
+        for m in masks_names_list:
+            if m not in hashes:
+                hashes[m] = cl_file.metadata[f'hash/{m}']
+        ell_hash = cl_file.metadata['hash/ell_hash']
+
+        w = {}
+        for tr1, tr2 in cl_file.get_tracer_combinations():
+            m1 = masks_names[tr1]
+            m2 = masks_names[tr2]
+            key = (m1, m2)
+            if (key in w) or (key[::-1] in w):
+                continue
+            # Build workspace hash (twopoint_fourier.py:387-395)
+            h1 = hashes[m1]
+            h2 = hashes[m2]
+            cache_key = h1 ^ ell_hash
+            if h2 != h1:
+                cache_key ^= h2
+
+            w[key] = str(cache.get_path(cache_key))
+
+        return w
+
+    def load_workspace_cache(self, dirname):
+        # Copied from twopoint_fourier.py
+        from .utils.nmt_utils import WorkspaceCache
+
+        if not dirname:
+            if self.rank == 0:
+                print("Not using an on-disc cache.  Set cache_dir to use one")
+            return {}
+
+        cache = WorkspaceCache(dirname)
+        return cache
+
+    def recover_NmtBin(self, cl_file):
+        # This function replicates `choose_ell_bins` in twopoint_fourier.py
+        from .utils.nmt_utils import build_MyNmtBin_from_binning_info
+
+        ell_min = cl_file.metadata['binning/ell_min']
+        ell_max = cl_file.metadata['binning/ell_max']
+        ell_spacing = cl_file.metadata['binning/ell_spacing']
+        n_ell = cl_file.metadata['binning/n_ell']
+
+        ell_bins = build_MyNmtBin_from_binning_info(ell_min, ell_max, n_ell,
+                                                    ell_spacing)
+
+        # Check that the binning is compatible with the one in the file
+        dtype = cl_file.get_data_types()[0]
+        trs = cl_file.get_tracer_combinations()[0]
+        ell_eff, _ = cl_file.get_ell_cl(dtype, *trs)
+        if not np.all(ell_bins.get_effective_ells() == ell_eff):
+            print(ell_bins.get_effective_ells())
+            print(ell_eff)
+            raise ValueError('The reconstructed NmtBin object is not ' +
+                             'compatible with the ells in the sacc file')
+
+        return ell_bins
+
+    def read_sacc(self):
+        # Loads a sacc file.
+        import sacc
+        f = self.get_input('twopoint_data_fourier')
+        two_point_data = sacc.Sacc.load_fits(f)
+
+        # Since NaMaster computes all terms (B-modes included). Keep all of
+        # them.
+
+        # # Remove the data types that we won't use for inference
```

Comment:
can you remove these old comments

Reply:
I've left the first one regarding to the B-modes. Shall I remove that one too?

Reply:
Nope, that's useful, let's leave that.